extractor/docker/docker.go

```diff
@@ -336,6 +336,14 @@ func (d DockerExtractor) ExtractFiles(layer io.Reader, filenames []string) (extr
 		// Determine if we should extract the element
 		extract := false
 		for _, s := range filenames {
+			// extract all files in target directory if last char is "/"(Separator)
+			if s[len(s)-1:] == "/" {
+				if filepath.Dir(s) == filepath.Dir(filePath) {
```

Comment:
Is filepath.Dir(s) necessary? Already checked.

```suggestion
				if s == filepath.Dir(filePath) {
```

Reply:
s : `var/lib/dpkg/status.d/`
filepath.Dir(s) : `var/lib/dpkg/status.d`

Reply:
`s[0:len(s)-2] == filepath.Dir(filePath)` looks mess.

Reply:
I got it. However, we already know `s` is a directory. It seems a little weird to apply filepath.Dir() to `s`. We may need a comment.

Reply:
How about `filepath.Clean`?

Reply:
>Dir calls Clean on the path and trailing slashes are removed.
https://golang.org/pkg/path/filepath/#Dir

Reply:
We must remove trailing slashes rather than extracting directory name. I wanna make it obvious.

Reply:
I fixed it.
I can change OSAnalyzer interface to following if you want.

```
type OSAnalyzer interface {
	Analyze(extractor.FileMap) (OS, error)
	RequiredFiles() []string
	RequiredDirs() []string
}
```